File: bank.py

import random

accounts = {}

def create_account():
    global accounts
    name = input("Enter User's name: ")
    balance = float(input("Enter initial balance: "))
    account_number = random.randint(1000000000, 9999999999)
    accounts[account_number] = {'name': name, 'balance': balance}
    print("Account created successfully!")
    print("Account number:", account_number)

def deposit_funds():
    account_number = int(input("Enter account number: "))
    if account_number in accounts:
        amount = float(input("Enter deposit amount: "))
        accounts[account_number]['balance'] += amount
        print("Deposit successful. New balance:", accounts[account_number]['balance'])
    else:
        print("Account not found.")

def withdraw_funds():
    account_number = int(input("Enter account number: "))
    if account_number in accounts:
        amount = float(input("Enter withdrawal amount: "))
        if amount <= accounts[account_number]['balance']:
            accounts[account_number]['balance'] -= amount
            print("Withdrawal successful. New balance:", accounts[account_number]['balance'])
        else:
            print("Insufficient balance.")
    else:
        print("Account not found.")

def main():
    while True:
        print("\nSelect the operation you want:")
        print("1. Create an account")
        print("2. Deposit funds")
        print("3. Withdraw funds")
        print("4. Exit")
        
        choice = input("Enter your choice: ")
        
        if choice == "1":
            create_account()
        elif choice == "2":
            deposit_funds()
        elif choice == "3":
            withdraw_funds()
        elif choice == "4":
            break
        else:
            print("Invalid choice. Please select a valid option.")

if __name__ == "__main__":
    main()

File: bank.js
const accounts = {};

function createAccount() {
  const name = prompt("Enter User's name:");
  const balance = parseFloat(prompt("Enter initial balance:"));
  const accountNumber = Math.floor(Math.random() * 9000000000) + 1000000000;
  accounts[accountNumber] = { name: name, balance: balance };
  console.log("Account created successfully!");
  console.log("Account number:", accountNumber);
}

function depositFunds() {
  const accountNumber = parseInt(prompt("Enter account number:"));
  if (accounts.hasOwnProperty(accountNumber)) {
    const amount = parseFloat(prompt("Enter deposit amount:"));
    accounts[accountNumber].balance += amount;
    console.log("Deposit successful. New balance:", accounts[accountNumber].balance);
  } else {
    console.log("Account not found.");
  }
}

function withdrawFunds() {
  const accountNumber = parseInt(prompt("Enter account number:"));
  if (accounts.hasOwnProperty(accountNumber)) {
    const amount = parseFloat(prompt("Enter withdrawal amount:"));
    if (amount <= accounts[accountNumber].balance) {
      accounts[accountNumber].balance -= amount;
      console.log("Withdrawal successful. New balance:", accounts[accountNumber].balance);
    } else {
      console.log("Insufficient balance.");
    }
  } else {
    console.log("Account not found.");
  }
}

function main() {
  while (true) {
    console.log("\nSelect the operation you want:");
    console.log("1. Create an account");
    console.log("2. Deposit funds");
    console.log("3. Withdraw funds");
    console.log("4. Exit");
    
    const choice = prompt("Enter your choice:");
    
    switch (choice) {
      case "1":
        createAccount();
        break;
      case "2":
        depositFunds();
        break;
      case "3":
        withdrawFunds();
        break;
      case "4":
        return;
      default:
        console.log("Invalid choice. Please select a valid option.");
    }
  }
}

main();
